main.cpp

```diff
@@ -70,29 +70,440 @@ write 3 UDTs below that EACH have:
+        }    
+
+        ++hours;
+    } 
+    std::cout << "Computer will crash in " << hours << " hours." << std::endl;
+    return hours;
+}
+
+bool ControlRoom::Computer::switchOnOff()
+{
+    powerState = !powerState;
+    if (powerState)
+    {
+        std::cout << "The computer is currently on."  << std::endl; 
+    } 
+    else
+    {
+        std::cout << "The computer is currently off."  << std::endl; 
+    }    
+    return powerState;
+}
+
+std::string ControlRoom::Computer::runSoftware(std::string programName)
+{
+    return programName;
+}
+
 /*
  copied UDT 3:
  */
 
+
+struct LiveRoom
+{
+    LiveRoom();                               
+    ~LiveRoom();    
+
+    int length = 26;
+    int width = 19;
+    int height = 4;
+    std::string wallMaterial = "cloth";
+    std::string floorMaterial = "wood";
+    bool lightsCurrentState = false; 
+    std::string name = "Studio A";
+
+    struct Musician
+    {
+        Musician();                               
+        ~Musician();   
+
+        std::string name, mainInstrument;
+        int yearsExperience, hourlyRate;
+
+        void callMusician();
+        bool createContract();
+        int totalHoursUnpaid(); 
+    };
+
+    void seatMusician(Musician musicianName, std::string thisName);
+    bool switchLights(); 
+    int calculateMusicianFee(int hours, bool receivesPublishingPercentage);
+};
+
+LiveRoom::LiveRoom()
+{
+    std::cout << "LiveRoom being constructed." << std::endl;
+}
+
+LiveRoom::~LiveRoom()
+{
+    std::cout << "LiveRoom being deconstructed." << std::endl;
+}
+
+LiveRoom::Musician::Musician() : name("John"), mainInstrument("Piano"), yearsExperience(10), hourlyRate(75)
+{
+    std::cout << "Musician being constructed." << std::endl;
+}
+
+LiveRoom::Musician::~Musician() 
+{
+    std::cout << "Musician being deconstructed." << std::endl;
+}
+
+int LiveRoom::calculateMusicianFee(int hours, bool receivesPublishingPercentage)
+{
+    int musicianFee = 0;    
+    for (int i = 0; i <= hours; ++i)
+    {
+        if(receivesPublishingPercentage == true)
+        {
+            musicianFee += 45;     // no overtime fee 
+        }
+        else
+        {
+            musicianFee += 115;
+            if(i % 10 == 0)
+            {
+                musicianFee += 30; // overtime fee
+            }                
+        }    
+    }
+    std::cout << "He is charging $" << musicianFee << " for this session." << std::endl;   
+    return musicianFee;
+}
+
+void LiveRoom::seatMusician(Musician musicianName, std::string thisName)
+{
+    musicianName.name = thisName;
+    std::cout << "Today we're enjoying the dulcet tones of " << musicianName.name  << std::endl;    
+}
+
+
+bool LiveRoom::switchLights()
+{
+    lightsCurrentState = !lightsCurrentState;
+    if (lightsCurrentState)
+    {
+        std::cout << "The lights are currently on."  << std::endl; 
+    } 
+    else
+    {
+        std::cout << "The lights are currently off."  << std::endl; 
+    }
```

Comment:
You can rewrite this as:
```
std::cout << "The lights are currently " << (lightsCurrentState ? "on." : "off.") << std::endl;
```
This makes use of a Ternary Expression. (check cppreference to learn about them)
Try it out. 
- [ ] revise

Reply:
Excellent, I used this is a few places to tighten the code a bit.